[Decomposition] Implement decomposition rules for Exp (#7489)
PennyLaneAI/pennylane: doc/releases/changelog-dev.md

  0: ──RX(-0.50)─╭●────────────┤  <Z>
  1: ────────────╰X──RY(-0.50)─┤
  ```
  

> * Decomposition rules compatible with the new graph-based decomposition system has been implemented
>   for `Exp`, specifically the following decompositions:
>   [(#7489)](https://github.com/PennyLaneAI/pennylane/pull/7489)
>   * Performs a Suzuki-Trotter decomposition when the `num_steps` keyword argument is specified.
>   * Decomposes to a `PauliRot` when the base is a single-term pauli word.

Reviewer: ```suggestion
* Decomposition rules compatible with the new graph-based decomposition system have been implemented
  for :class:`~pennylane.ops.Exp`. Specifically, the following decompositions have been added:
  * Suzuki-Trotter decomposition when the `num_steps` keyword argument is specified.
  * Decomposition to a :class:`~pennylane.PauliRot` when the base is a single-term Pauli word.
  [(#7489)](https://github.com/PennyLaneAI/pennylane/pull/7489)
```

* Decomposition rules compatible with the new graph-based decomposition system has been implemented
  for `Exp`, specifically the following decompositions:
  [(#7489)](https://github.com/PennyLaneAI/pennylane/pull/7489)

>   * Performs a Suzuki-Trotter decomposition when the `num_steps` kwarg is specified.

Reviewer: ```suggestion
  * Performs a Suzuki-Trotter decomposition when the `num_steps` keyword argument is specified.
```
